Source/WebCore/platform/graphics/ImageSource.cpp

```diff
@@ -291,14 +302,20 @@ void ImageSource::cachePlatformImageAtIndex(PlatformImagePtr&& platformImage, si
         return;
 
     // Move the new image to the cache.
-    frame.m_nativeImage = NativeImage::create(WTFMove(platformImage));
+    frame.m_nativeImage = WTFMove(nativeImage);
+    frame.m_cachedDecodedImage.reset();
```

Comment:
I think we never hit the case where m_cachedDecodedImage is not nullptr. Maybe we need to `ASSERT(!frame.m_cachedDecodedImage);` instead.